BrickKit/bricks/src/main/java/com/wayfair/brickkit/BrickDataManager.java

```diff
@@ -94,21 +93,20 @@ private void setRecyclerView(@NonNull RecyclerView recyclerView, int orientation
 
         this.recyclerView = recyclerView;
         this.recyclerView.setAdapter(brickRecyclerAdapter);
-        if (itemDecoration != null) {
-            this.recyclerView.removeItemDecoration(itemDecoration);
+        while (recyclerView.getItemDecorationCount() > 0) {
+            recyclerView.removeItemDecorationAt(0);
         }
-        this.itemDecoration = new BrickRecyclerItemDecoration(this);
-        this.recyclerView.addItemDecoration(itemDecoration);
+        this.recyclerView.addItemDecoration(new BrickRecyclerItemDecoration(this));
         this.recyclerView.setItemAnimator(new AvoidFlickerItemAnimator());
 
         GridLayoutManager gridLayoutManager = new WFGridLayoutManager(recyclerView.getContext(), orientation);
         gridLayoutManager.setSpanSizeLookup(new BrickSpanSizeLookup(recyclerView.getContext(), this));
         recyclerView.setLayoutManager(gridLayoutManager);
 
-        LinkedList<BaseBrick> items = getRecyclerViewItems();
-        int itemCount = items.size();
+        LinkedList<BaseBrick> recyclerViewItems = getRecyclerViewItems();
```

Comment:
can this just be a plain List on the left hand side?

Reply:
Yeah, I was gonna do more of the LinkedList -> List API changes all at once

Reply:
Did the internal ones